ddtrace/contrib/pytest/plugin.py

```diff
@@ -110,7 +110,7 @@ def pytest_sessionstart(session):
         tracer_filters = pin.tracer._filters
         if not any(isinstance(tracer_filter, TraceCiVisibilityFilter) for tracer_filter in tracer_filters):
             tracer_filters += [TraceCiVisibilityFilter()]
-            pin.tracer.configure(settings={"FILTERS": tracer_filters})
+            pin.tracer.configure(settings={"FILTERS": tracer_filters}, enabled=ddtrace.config.pytest.trace_enabled)
```

Comment:
🤔 was `configure` causing enabled to be set to True here? I'm confused as to how pytest was enabling a tracer despite DD_TRACE_ENABLED=false

Reply:
So I realize this might actually fix the underlying issue from #4179, I can't seem to recreate the exact issue where the pytest plugin is enabling a tracer by importing `ddtrace`. I've tried adding a `@tracer.wrap()` decorator and running both `pytest` and `pytest_bdd` test cases, to no avail 😖 

However, our pytest plugin always runs a `tracer.configure(...)` at the start of every test session in the `pytest_sesionstart()` hook, and explicitly setting `tracer.enabled` to match `DD_TRACE_ENABLED` should fix the side effect.